testcases/PowerNVDump.py

```diff
@@ -865,13 +865,48 @@ def runTest(self):
                 boot_type = self.kernel_crash(crash_type="hmc")
                 self.verify_dump_file(boot_type)
 
+class KernelCrash_KdumpDLPAR(PowerNVDump, testcases.OpTestDlpar.OpTestDlpar):
+
+    # This test verifies kdump/fadump after cpu and memory add/remove.
+    # cpu_resource and mem_resource must be defined in ~/.op-test-framework.conf.
+    # cpu_resource - max number of CPU
+    # mem_resource - max memory in MB
+    # Ex: cpu_resource=4
+    #     mem_resource=2048
+
+    def runTest(self):
+        self.extended = {'loop':0,'wkld':0,'smt':8}
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        self.setup_test()
+        self.AddRemove("proc","--procs","r",self.cpu_resource)
+        print("=============== Testing kdump/fadump after cpu remove ===============")
+        boot_type = self.kernel_crash()
+        self.verify_dump_file(boot_type)
+        self.setup_test()
+        self.AddRemove("proc","--procs","a",self.cpu_resource)
+        print("=============== Testing kdump/fadump after cpu add ===============")
+        boot_type = self.kernel_crash()
+        self.verify_dump_file(boot_type)
+        self.setup_test()
+        self.AddRemove("mem", "-q", "a", self.mem_resource)
+        print("=============== Testing kdump/fadump after memory add ===============")
+        boot_type = self.kernel_crash()
+        self.verify_dump_file(boot_type)
+        self.setup_test()
+        self.AddRemove("mem", "-q", "r", self.mem_resource)
+        print("=============== Testing kdump/fadump after memory remove ===============")
+        boot_type = self.kernel_crash()
+        self.verify_dump_file(boot_type)
+
```

Comment:
A lot of code is repeated in the above section. Can this be rewritten may be using for loop to make it compact and easy to read. Something on the following lines (sudo code only, not compile/syntax tested)

for comp in ['proc', 'mem']:
    for operation in ['r', 'a']:
         self.setup_test()
         self.addRemove(comp, ....)
         boot_type = self.kernel_crash()
         self.verify_dump_file(boot_type)

Possibly few checks may be required to construct the required parameters to be passed to AddRemove() function